### DISCUSSION TITLE
Option for Isometric Grid

### DISCUSSION THREAD

**Discussion by bhargavmodak** [excalidraw/excalidraw#5345] [General]
I wasn't sure where I could put in suggestions or requests for Excalidraw, so I came to the github repo. 

I want to ask if it is feasible to add the option to add an isometric grid (like in Inkscape) to Excalidraw. One workaround I have been using is by creating a low opacity grid myself, like so: 

![image](https://user-images.githubusercontent.com/82528318/174985006-8ea7e46a-a5fe-4808-814b-b90cd6d4b8d6.png)

But it quickly takes up a lot of the space on the board, slowing down the response time. 

Usually I use it to draw a quick design of a room or place I use in my artistic creations, like stories. It is useful to have a visual guide to write the story better. I could use Inkscape for the drawings, but sometimes I want to create some quick 3d models for the rooms and such. 

The hand drawn feel of Excalidraw is much useful to sell the e-stories I write. 

I would do a pull request myself, but I have no experience with Typescript yet.

**Comment by takiyao** [comment]
I am also interested in something like this. The best I could do was run the app locally, remove the background color, and add the isometric image as a background. Unfortunately, that means the background exists outside the canvas and doesn't move or scale with the rest of the content.

**Comment by nstefan13** [comment]
Hey! I recently did a [pull request](https://github.com/excalidraw/excalidraw/pull/9893) where I added a isometric dot grid. If you'd like to give it a try, the most recent version has been deployed [here](https://excalidraw-9nl89nkl0-excalidraw.vercel.app/).

Note: In order to change to a dot grid or an isometric dot grid, you first have to toggle the default grid (which can be done using the **Toggle Grid** command from the command pallet). After that, you simply open the main menu and select your desired grid from the **Grid Type** setting (which is right above the **Canvas background** setting). Below is a video of how it:


https://github.com/user-attachments/assets/bf613317-2b38-4765-9b47-f0461dae1756

**Reply by takiyao** [reply]
I tried it and I liked it. There's some odd snapping though where the lines are snapping to other points that are not visible.


<img width="1013" height="834" alt="image" src="https://github.com/user-attachments/assets/f667cc3b-103d-40f7-b801-7244f403e81b" />

**Reply by bhargavmodak** [reply]
> I tried it and I liked it. There's some odd snapping though where the lines are snapping to other points that are not visible.

This is because it isn't a new grid with new mapping, if you switch back to default you'll the the lines are actually mapping to the rectangular grid underneath
